### PR TITLE
Update train_with_distillation.py

### PR DESCRIPTION
**Pull request by david-0429**
기존 Train_loss : KD loss가 없는 그냥 CE loss(즉, Total_loss는 존재하지 않음)
바뀐 점 : +Total_loss, +CE_loss(기존 Train_loss), +KD_loss 추가
              wandb에다가 Total_loss, CE_loss, KD_loss 추가

단, test는 건들지 않음

problem : KD_loss의 loss_distill을 계산 하는 부분이 아리송함(batch 계산)

이게 맞는건지 잘 모르겠어서 한번만 검토해줘,,,,,,ㅠㅠ

### CODE REVIEW

File: CIFAR-100/train_with_distillation.py [hunk 1]
```diff
@@ -129,9 +134,10 @@ def train_with_distill(d_net, epoch, alpha):
         b_idx = batch_idx
 
     print('Train \t Time Taken: %.2f sec' % (time.time() - epoch_start_time))
-    print('Loss: %.3f | Acc: %.3f%% (%d/%d)' % (train_loss / (b_idx + 1), 100. * correct / total, correct, total))
+    print('total_loss: %.3f | KD Loss: %.3f | CE_Loss: %.3f | Acc: %.3f%% (%d/%d)' % (total_loss / (b_idx + 1), KD_loss / (b_idx + 1), CE_loss / (b_idx + 1), 100. * correct / total, correct, total))
```
Comment (krenerd) [inline CIFAR-100/train_with_distillation.py:137]:
```suggestion
    print('total_loss: %.3f | Distill Loss: %.3f | CE_Loss: %.3f | Acc: %.3f%% (%d/%d)' % (total_loss / (b_idx + 1), distill_loss / (b_idx + 1), CE_loss / (b_idx + 1), 100. * correct / total, correct, total))
```

File: CIFAR-100/train_with_distillation.py [hunk 2]
```diff
@@ -129,9 +134,10 @@ def train_with_distill(d_net, epoch, alpha):
...
 
-    return train_loss / (b_idx + 1)
+
+    return total_loss / (b_idx + 1), CE_loss / (b_idx + 1), KD_loss / (b_idx + 1)
```
Comment (krenerd) [inline CIFAR-100/train_with_distillation.py:140]:
```suggestion
    return total_loss / (b_idx + 1), CE_loss / (b_idx + 1), distill_loss / (b_idx + 1)
```
LGTM, 
- rename `KD_loss` -> `distill_loss`

File: CIFAR-100/train_with_distillation.py [hunk 3]
```diff
@@ -116,11 +118,14 @@ def train_with_distill(d_net, epoch, alpha):
...
         loss = loss_CE + alpha * (loss_distill.sum()) / batch_size / 1000
+        loss_distill = loss_distill.sum() / batch_size / 1000
 
         loss.backward()
         optimizer.step()
 
-        train_loss += loss_CE.item()
+        total_loss += loss.item()
+        CE_loss += loss_CE.item()
+        KD_loss += loss_distill.item()
```
Comment (krenerd) [inline CIFAR-100/train_with_distillation.py:128]:
```suggestion
        distill_loss += loss_distill.item()
```

File: CIFAR-100/train_with_distillation.py [hunk 4]
```diff
@@ -101,7 +101,9 @@ def train_with_distill(d_net, epoch, alpha):
     d_net.s_net.train()
     d_net.t_net.train()
 
-    train_loss = 0
+    total_loss = 0
+    KD_loss = 0
```
Comment (krenerd) [inline CIFAR-100/train_with_distillation.py:105]:
```suggestion
    distill_loss += 0
```

File: CIFAR-100/train_with_distillation.py [hunk 5]
```diff
@@ -172,8 +178,8 @@ def test(net):
         optimizer = optim.SGD([{'params': s_net.parameters()}, {'params': d_net.Connectors.parameters()}],
                               lr=args.lr / 100, momentum=args.momentum, weight_decay=args.weight_decay, nesterov=True)
 
-    train_loss = train_with_distill(d_net, epoch, args.alpha)
+    total_loss, CE_loss, KD_loss = train_with_distill(d_net, epoch, args.alpha)
```
Comment (krenerd) [inline CIFAR-100/train_with_distillation.py:181]:
```suggestion
    total_loss, CE_loss, distill_loss = train_with_distill(d_net, epoch, args.alpha)
```

File: CIFAR-100/train_with_distillation.py [hunk 6]
```diff
@@ -172,8 +178,8 @@ def test(net):
...
     test_loss, accuracy = test(s_net)
-    wandb.log({"epoch/val_acc": accuracy, "epoch/val_loss": test_loss, "epoch/trn_loss": train_loss, "epoch": epoch})
+    wandb.log({"epoch/val_acc": accuracy, "epoch/val_loss": test_loss, "epoch/total_loss": total_loss, "epoch/CE_loss": CE_loss, "epoch/KD_loss": KD_loss, "epoch": epoch})
```
Comment (krenerd) [inline CIFAR-100/train_with_distillation.py:183]:
```suggestion
    wandb.log({"epoch/val_acc": accuracy, "epoch/val_loss": test_loss, "epoch/total_loss": total_loss, "epoch/CE_loss": CE_loss, "epoch/Distillation_loss": distill_loss, "epoch": epoch})
```

**Review by sieu-n — CHANGES_REQUESTED** [review]
Make code prettier, remove /1000 and set default value of alpha smaller.

Were you able to successfully run the code?

File: CIFAR-100/train_with_distillation.py [hunk 7]
```diff
@@ -116,11 +118,14 @@ def train_with_distill(d_net, epoch, alpha):
         loss_CE = criterion_CE(outputs, targets)
 
         loss = loss_CE + alpha * (loss_distill.sum()) / batch_size / 1000
+        loss_distill = loss_distill.sum() / batch_size / 1000
```
Comment (sieu-n) [inline CIFAR-100/train_with_distillation.py]:
```suggestion
        loss_CE = criterion_CE(outputs, targets)
        loss_distill = loss_distill.sum() / batch_size

        loss = loss_CE + alpha * loss_distill
```
Make code prettier, remove `/1000` and set default value of alpha smaller.

**Comment by david-0429** [comment]
good ^^
